homeassistant/components/vizio/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for Vizio."""
+
+import logging
+from typing import Any, Dict
+
+from pyvizio import VizioAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_DEVICE_CLASS,
+    CONF_HOST,
+    CONF_NAME,
+)
+from homeassistant.core import callback
+
+from . import validate_auth
+from .const import (
+    CONF_VOLUME_STEP,
+    DEFAULT_DEVICE_CLASS,
+    DEFAULT_NAME,
+    DEFAULT_VOLUME_STEP,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+VIZIO_SCHEMA = {
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+    vol.Required(CONF_HOST, default=""): str,
```

Comment:
In this case, it was a mistake from testing. However I am curious, I have other parameters that are optional in the config because there is a default, but are required to have some value. Does it not make sense to make them required in the user config flow but to provide defaults so that they are prefilled but have to have a value in order to pass?

Reply:
Ok, yeah that's ok. Since the user can change the default in this case, it's a valid approach.